core/src/main/scala/org/apache/spark/internal/config/UI.scala

```diff
@@ -229,4 +229,11 @@ private[spark] object UI {
     .stringConf
     .transform(_.toUpperCase(Locale.ROOT))
     .createWithDefault("LOCAL")
+
+  val UI_SQL_GROUP_SUB_EXECUTION_ENABLED = ConfigBuilder("spark.ui.sql.group.sub.execution.enabled")
+    .doc("Whether to group sub executions together in SQL UI when they belong to the same " +
+      "root execution")
+    .version("3.4.0")
+    .booleanConf
+    .createWithDefault(false)
```

Comment:
shall we turn it on by default?

Reply:
yes, we should

Reply:
It's okay to enable this by default. Please update the PR.

Reply:
done